src/Files.Backend/Helpers/PathHelpers.cs

```diff
@@ -0,0 +1,39 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Files.Backend.Helpers
+{
+	public static class PathHelpers
+	{
+		public static string FormatName(string path = null)
+		{
+			string fileName;
+
+			// Network Share path
+			if (Path.GetPathRoot(path) == path && path.StartsWith(@"\\"))
+			{
+				fileName = path.Substring(path.LastIndexOf(@"\") + 1);
+			}
+			// Drive path
+			else if (Path.GetPathRoot(path) == path)
```

Comment:
`Path.GetPathRoot(path)` can be put inside a variable to not be evaluated twice.